ulib/experimental/Steel.SLDump.fsti

```diff
@@ -0,0 +1,23 @@
+module Steel.SLDump
+
+include Steel.SLDump.Base
+include Steel.Effect
+include Steel.Effect.Atomic
+
+val sldump'
+  (#opened: _)
+  (p: vprop)
+  (q: vprop)
+  (text: string)
+  (sq: squash (sldump_prop text p q))
+  (_: unit)
+: SteelGhost unit opened p (fun _ -> guard_vprop q) (fun _ -> True) (fun h _ h' -> h' p == h p) // TODO: replace with frame_equalities p h h'?
+
+val sldump
+  (#opened: _)
+  (#[@@@framing_implicit] p: vprop)
+  (#[@@@framing_implicit] q: vprop)
+  (text: string)
+  (#[@@@framing_implicit] sq: squash (sldump_prop text p q))
```

Comment:
I think you could put this squash before the `text` argument, and remove the need for the trailing unit.

Reply:
I don't think so: I made `sldump_prop` depends on `text`, so that `solve_sldump_prop` (the tactic to solve `sldump_prop`) can print that additional `text`. If I were to remove that dependency, then I wouldn't know how `solve_sldump_prop` would grab `text`.

Reply:
Ah right, sorry, I was still thinking of sldump_prop as being in the requires, and not as an argument